python/jupytergis_lab/jupytergis_lab/notebook/gis_document.py

```diff
@@ -72,7 +72,9 @@ def __init__(
         ydoc = Doc()
 
         super().__init__(
-            comm_metadata=dict(ymodel_name="@jupytergis:widget", **comm_metadata),
+            comm_metadata=dict(
+                ymodel_name="@jupytergis:widget", cwd=os.getcwd(), **comm_metadata
```

Comment:
Do we really need this? cc. @davidbrochart for awareness

Reply:
It is required to be able to add local layer from the widget buttons.
It could probably be removed if we don't use file path relative to the JGIS project file.

Reply:
> It is required to be able to add local layer from the widget buttons

Once the JGIS file is created, we know its "cwd" is the path to the file, right? Since we know that path in the front-end already, we may not need to pass this info from Python?

Reply:
Yes, it is not used if a path is provided.
But we want to create a `GISDocument` without file, no ?
```python
doc = GISDocument()
```

Reply:
In that case, we can assume the "cwd" is the Notebook path, since we have the notebook tracker handy?

Reply:
Yes, I'll update it

Reply:
Actually I remember why I did not do that the first time.
Should it also work in console ? Is there a way to get the path from frontend in that case ?

Reply:
arf

Reply:
All this situation is not great. We need a better approach but I don't have a good quick solution to suggest.

See https://github.com/geojupyter/jupytergis/pull/426#issuecomment-2631209276

Reply:
Are the widgets supported in console ? I don't know how to display it in console.
If not, we don't need the path, it is useful for the buttons only, not for the API I think.

Reply:
That should work in most cases. It's only when people changed did `os.chdir` that it will break this. But I guess we can live with this.